File: query_tenants.js
const { Client } = require('pg');

async function main() {
    const client = new Client({
        host: 'localhost',
        port: 5432,
        database: 'fralib_db',
        user: 'postgres',
        password: 'fralib2024'
    });

    try {
        await client.connect();
        console.log('Connected to PostgreSQL!');

        // Get tenants
        const usersResult = await client.query('SELECT id, email, plano, nicho FROM users ORDER BY id');
        const users = usersResult.rows;
        console.log(`\nFound ${users.length} tenants\n`);

        for (const user of users) {
            console.log('='.repeat(80));
            console.log(`TENANT: ${user.email}`);
            console.log(`  ID: ${user.id} | Plano: ${user.plano} | Nicho: ${user.nicho}`);
            console.log('');

            // SDR settings
            const sdrResult = await client.query(
                'SELECT config_value FROM user_configs WHERE user_id = $1 AND config_key = $2',
                [user.id, 'sdr_settings_v1']
            );

            if (sdrResult.rows.length > 0) {
                const sdr = JSON.parse(sdrResult.rows[0].config_value);
                console.log('  SDR_SETTINGS_V1:');
                console.log(`    agent_name: ${sdr.agent_name || 'N/A'}`);
                console.log(`    limits: ${JSON.stringify(sdr.limits || {})}`);
                console.log(`    schedule: ${JSON.stringify(sdr.outbound_schedule || {})}`);
                console.log(`    blocked_actions: ${JSON.stringify(sdr.blocked_actions || [])}`);
                console.log(`    allowed_actions: ${JSON.stringify(sdr.allowed_actions || [])}`);
                console.log(`    handoff: ${JSON.stringify(sdr.handoff || {})}`);
            } else {
                console.log('  SDR_SETTINGS_V1: NOT CONFIGURED');
            }
            console.log('');

            // Pipeline state
            const pipeResult = await client.query(
                'SELECT config_value FROM user_configs WHERE user_id = $1 AND config_key = $2',
                [user.id, 'pipeline_state']
            );

            if (pipeResult.rows.length > 0) {
                console.log('  PIPELINE_STATE:');
                const pipeline = JSON.parse(pipeResult.rows[0].config_value);
                console.log(`    ${JSON.stringify(pipeline)}`);
            } else {
                console.log('  PIPELINE_STATE: NOT CONFIGURED');
            }
            console.log('');

            // Lead supply config
            const lscResult = await client.query(
                'SELECT config_value FROM user_configs WHERE user_id = $1 AND config_key = $2',
                [user.id, 'lead_supply_config']
            );

            if (lscResult.rows.length > 0) {
                console.log('  LEAD_SUPPLY_CONFIG:');
                const lsc = JSON.parse(lscResult.rows[0].config_value);
                console.log(`    ${JSON.stringify(lsc, null, 4)}`);
            } else {
                console.log('  LEAD_SUPPLY_CONFIG: NOT CONFIGURED');
            }
            console.log('');

            // Lead counts
            const leadResult = await client.query(
                'SELECT status, COUNT(*) as count FROM leads WHERE user_id = $1 GROUP BY status',
                [user.id]
            );

            console.log('  LEAD STATUS COUNTS:');
            let pendingSdr = 0;
            let pendingWpp = 0;
            for (const row of leadResult.rows) {
                console.log(`    ${row.status}: ${row.count}`);
                if (row.status === 'pending_sdr_send') pendingSdr = parseInt(row.count);
                if (row.status === 'pendente_wpp') pendingWpp = parseInt(row.count);
            }
            console.log(`  -> Leads stuck at pending_sdr_send: ${pendingSdr}`);
            console.log(`  -> Leads stuck at pendente_wpp: ${pendingWpp}`);
            console.log('');
        }

        await client.end();
    } catch (err) {
        console.error('Error:', err);
        process.exit(1);
    }
}

main();

File: query_tenants.py
# -*- coding: utf-8 -*-
import psycopg2
import json
import sys

def main():
    # Force UTF-8 encoding for all streams
    import io
    sys.stdout = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8', errors='replace')
    sys.stderr = io.TextIOWrapper(sys.stderr.buffer, encoding='utf-8', errors='replace')

    # Try connection
    conn = psycopg2.connect(
        host="localhost",
        port=5432,
        database="fralib_db",
        user="postgres",
        password="fralib2024"
    )
    print("Connected successfully!", flush=True)

    cur = conn.cursor()

    # Get tenants
    cur.execute("SELECT id, email, plano, nicho FROM users ORDER BY id")
    rows = cur.fetchall()
    print(f"\nFound {len(rows)} tenants", flush=True)

    # Store results for structured output
    all_tenants = []

    for row in rows:
        tenant_id, email, plano, nicho = row
        tenant_data = {
            'id': tenant_id,
            'email': email,
            'plano': plano,
            'nicho': nicho,
            'sdr_settings': None,
            'pipeline_state': None,
            'lead_supply_config': None,
            'lead_counts': {}
        }

        # SDR settings
        cur.execute("""
            SELECT config FROM user_configs
            WHERE user_id = %s AND key = 'sdr_settings_v1'
        """, (tenant_id,))
        sdr_row = cur.fetchone()

        if sdr_row:
            config = sdr_row[0]
            if isinstance(config, str):
                try:
                    config = json.loads(config)
                except:
                    pass
            tenant_data['sdr_settings'] = config

        # Pipeline state
        cur.execute("""
            SELECT config FROM user_configs
            WHERE user_id = %s AND key = 'pipeline_state'
        """, (tenant_id,))
        pipeline_row = cur.fetchone()

        if pipeline_row:
            config = pipeline_row[0]
            if isinstance(config, str):
                try:
                    config = json.loads(config)
                except:
                    pass
            tenant_data['pipeline_state'] = config

        # Lead supply config
        cur.execute("""
            SELECT config FROM user_configs
            WHERE user_id = %s AND key = 'lead_supply_config'
        """, (tenant_id,))
        lsc_row = cur.fetchone()

        if lsc_row:
            config = lsc_row[0]
            if isinstance(config, str):
                try:
                    config = json.loads(config)
                except:
                    pass
            tenant_data['lead_supply_config'] = config

        # Lead counts
        cur.execute("""
            SELECT status, COUNT(*) as count
            FROM leads
            WHERE user_id = %s
            GROUP BY status
        """, (tenant_id,))
        lead_counts = cur.fetchall()
        for status, count in lead_counts:
            tenant_data['lead_counts'][status] = count

        all_tenants.append(tenant_data)

    # Output JSON for structured processing
    print("\n=== STRUCTURED DATA ===", flush=True)
    print(json.dumps(all_tenants, ensure_ascii=False, indent=2, default=str), flush=True)

    conn.close()

if __name__ == "__main__":
    main()
